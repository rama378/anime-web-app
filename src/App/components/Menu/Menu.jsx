import { useNavigate } from 'react-router-dom';
import Button from '../../common/components/Button/Button';
import { PAGE } from '../../common/constants/constant';
import './Menu.css';

const Menu = () => {
	const navigate = useNavigate();
	
	const openLogInPage = () => {
		navigate(PAGE.LOGIN);
	};

	const openSignUpPage = () => {
		navigate(PAGE.SIGNUP);
	};

	return (
		<div>
			<Button onClick={openSignUpPage} style="clear">Sign Up</Button>
			<Button onClick={openLogInPage}>Log In</Button>
		</div>
	);
};

export default Menu;
