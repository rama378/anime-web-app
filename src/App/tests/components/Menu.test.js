import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import Menu from '../../components/Menu/Menu';

const renderer = new ShallowRenderer();

describe('<Menu />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(
      (
        <BrowserRouter>
          <Menu />
        </BrowserRouter>
      )
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
