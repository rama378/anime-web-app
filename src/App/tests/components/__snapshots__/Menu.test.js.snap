// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render and match the snapshot 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "createHref": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Menu />
</Router>
`;
